drivers/i2c/i2c_ll_stm32.c

```diff
@@ -93,6 +93,9 @@ static int i2c_stm32_init(struct device *dev)
 	const struct i2c_stm32_config *cfg = DEV_CFG(dev);
 	u32_t bitrate_cfg;
 	int ret;
+#ifdef CONFIG_I2C_STM32_INTERRUPT
+	struct i2c_stm32_data *data = DEV_DATA(dev);
+#endif
```

Comment:
remove these 3 lines, and below on line 110, have a if (IF_ENABLED(CONFIG_I2C_STM32_INTERRUPT) {
    struct i2c_stm32_data *data = DEV_DATA(dev);

    < ... what's currently done in 111 and 112 ... >
}

That way you won't need any of the ifdefsf anymore in this function, it's nicer to read then.